.buildkite/scripts/common2.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+set -euo pipefail
+
+source .buildkite/scripts/retry.sh
+
+if [[ -z "${WORKSPACE-""}" ]]; then
+    WORKSPACE=$(git rev-parse --show-toplevel)
+    export WORKSPACE
+fi
+if [[ -z "${SETUP_MAGE_VERSION-""}" ]]; then
+    SETUP_MAGE_VERSION="1.14.0"
+fi
+if [[ -z "${SETUP_GVM_VERSION-""}" ]]; then
+    SETUP_GVM_VERSION="v0.5.0" # https://github.com/andrewkroh/gvm/issues/44#issuecomment-1013231151
+fi
+BEAT_VERSION=$(grep -oE '[0-9]+\.[0-9]+\.[0-9]+(\-[a-zA-Z]+[0-9]+)?' "${WORKSPACE}/version/version.go")
+export BEAT_VERSION
+
+getOSOptions() {
+  case $(uname | tr '[:upper:]' '[:lower:]') in
+    linux*)
+      export AGENT_OS_NAME=linux
+      ;;
+    darwin*)
+      export AGENT_OS_NAME=darwin
+      ;;
+    msys*)
+      export AGENT_OS_NAME=windows
+      ;;
+    *)
+      export AGENT_OS_NAME=notset
+      ;;
+  esac
+  case $(uname -m | tr '[:upper:]' '[:lower:]') in
+    aarch64*)
+      export AGENT_OS_ARCH=arm64
+      ;;
+    arm64*)
+      export AGENT_OS_ARCH=arm64
+      ;;
+    amd64*)
+      export AGENT_OS_ARCH=amd64
+      ;;
+    x86_64*)
+      export AGENT_OS_ARCH=amd64
+      ;;
+    *)
+      export AGENT_OS_ARCH=notset
+      ;;
+  esac
+}
+
+# Wrapper function for executing mage
+install_mage() {
+    go version
+    if ! [ -x "$(type -P mage | sed 's/mage is //g')" ];
+    then
+        echo "installing mage ${SETUP_MAGE_VERSION}"
+        make mage
+    fi
+    pushd "$WORKSPACE"
+}
+
+install_go() {
```

Comment:
Do we still need this `install_go`, `install_mage` functions, given we are running with custom images?

Reply:
For MacOS

Reply:
Let's remove it for now. Hopefully, we will find a way to preinstall it for MacOS